src/Compilers/Core/Portable/RuntimeCapability.cs

```diff
@@ -0,0 +1,88 @@
+        /// </summary>
+        UnmanagedSignatureCallingConvention = 5,
+
+        /// <summary>
+        /// Indicates that this version of runtime supports virtual static members of interfaces.
+        /// </summary>
+        VirtualStaticsInInterfaces = 6,
+    }
+
+    internal static class RuntimeCapabilityHelpers
+    {
+        public static bool RuntimeSupportsCapability(IAssemblySymbolInternal assembly, RuntimeCapability capability)
+        {
+            switch (capability)
+            {
+                case RuntimeCapability.ByRefFields:
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__ByRefFields);
+
+                case RuntimeCapability.CovariantReturnsOfClasses:
+                    // check for the runtime feature indicator and the required attribute.
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__CovariantReturnsOfClasses) &&
+                           assembly.GetSpecialType(SpecialType.System_Runtime_CompilerServices_PreserveBaseOverridesAttribute) is { TypeKind: TypeKind.Class };
+
+                case RuntimeCapability.DefaultImplementationsOfInterfaces:
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__DefaultImplementationsOfInterfaces);
+
+                case RuntimeCapability.NumericIntPtr:
+                    // CorLibrary should never be null, but that invariant is broken in some cases for MissingAssemblySymbol.
+                    // Tracked by https://github.com/dotnet/roslyn/issues/61262
+                    return assembly.CorLibrary is not null &&
+                           RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__NumericIntPtr);
+
+                case RuntimeCapability.UnmanagedSignatureCallingConvention:
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__UnmanagedSignatureCallingConvention);
+
+                case RuntimeCapability.VirtualStaticsInInterfaces:
+                    return RuntimeSupportsFeature(assembly, SpecialMember.System_Runtime_CompilerServices_RuntimeFeature__VirtualStaticsInInterfaces);
+
+                default:
+                    return false;
+            }
+        }
+
+        private static bool RuntimeSupportsFeature(IAssemblySymbolInternal assembly, SpecialMember feature)
+        {
+            Debug.Assert((SpecialType)SpecialMembers.GetDescriptor(feature).DeclaringTypeId == SpecialType.System_Runtime_CompilerServices_RuntimeFeature);
+            return assembly.IsStaticClass(assembly.GetSpecialType(SpecialType.System_Runtime_CompilerServices_RuntimeFeature)) &&
+                   assembly.GetSpecialTypeMember(feature) is not null;
```

Comment:
vb and C# are slightly different how they test if something is a static class.  hence why we defer to the language to make the determination here.